microsetta_public_api/utils/_utils.py

```diff
@@ -19,3 +21,64 @@ def check_missing_ids(missing_ids, alpha_metric, type_):
                        error=404, text=f"Sample ID(s) not found for "
                                        f"{type_}: {alpha_metric}"), \
                404
+
+
+_data_table = namedtuple('DataTable', ['data', 'columns'])
+
+
+class DataTable(_data_table):
+    """A convenience class for constructing jQuery a DataTable in python
+    that is JSON serializable.
+
+    A DataTable should have data entries and a list of columns that are in
+    the entries.
+
+    Attributes
+    ----------
+
+    data : list of DataEntry
+        The entries that comprise the DataTable
+    columns : list of str
+        The names of the columns in the DataTable
+
+    """
+    def to_dict(self):
+        data = self.data.copy()
```

Comment:
it looks like this function is first doing a full copy, and then walking back over that same data? I think the amount of effort needed here could be effectively halved with:

```python
dict_ = self._asdict()
dict_['data'] = [entry.to_dict() for entry in enumerate(self.data)]
return dict_
```